### PR TITLE
Electron gets smarter

### PR DESCRIPTION
**Pull request by nuft**
Some little improvements of the electron:
- add timeout for reaching target
- comunicate state through LED blink pattern

This change is tested and working on hardware.

### CODE REVIEW

File: electron-firmware/src/uavcan/node.cpp [hunk 1]
```diff
@@ -60,14 +73,35 @@ void main(unsigned int id, const char* name)
...
+        }
+
+        if (electron_state == RUNNING) {
+            if (data_packet_start_signal_received() && !front_hall_sensor()) {
+                motor_voltage_set(voltage);
+            } else {
+                motor_voltage_set(0.0f);
+            }
+
+            if (front_hall_sensor() && timed_out) {
```
Comment (antoinealb) [inline electron-firmware/src/uavcan/node.cpp]: Shouldn't it be `front_hall || timed_out` ?

File: electron-firmware/src/uavcan/node.h [hunk 2]
```diff
@@ -5,6 +5,14 @@
 extern "C" {
 #endif
 
+enum {
+    INIT,
+    READY,
+    RUNNING,
+    ARRIVED
+};
```
Comment (SyrianSpock) [inline electron-firmware/src/uavcan/node.h]: Could be a `enum class ElectronState {...}` for type safety, then you'll need to use `ElectronState::INIT`

File: electron-firmware/src/uavcan/node.cpp (hunk 1 above)
Comment (SyrianSpock) [reply electron-firmware/src/uavcan/node.cpp]: I thought the idea was to continue climbing until timing out, but I'm not sure anymore what we're trying to achieve :thinking:

Comment (nuft) [reply electron-firmware/src/uavcan/node.cpp]: The idea was that it continues to try to climb until the timeout.
I realize now that it should only be `if (timed_out)`.

File: electron-firmware/src/uavcan/node.h (hunk 2 above)
Comment (nuft) [reply electron-firmware/src/uavcan/node.h]: I want it to be usable from `main.c` and I'm too lazy to make it a `main.cpp`.

File: electron-firmware/src/uavcan/node.cpp (hunk 1 above)
Comment (nuft) [reply electron-firmware/src/uavcan/node.cpp]: changed now. I added a TIMED_OUT state that should allow to define success only when in ARRIVED state. We could add a message back to the robot for exact score counting.

File: electron-firmware/src/uavcan/node.h (hunk 2 above)
Comment (SyrianSpock) [reply electron-firmware/src/uavcan/node.h]: fair enough

**Pull request merged**